File: src/main/webapp/src/app/models/runtime-configuration.ts
export interface RuntimeConfigurationIntf {
  tagsInField?: number;
  memorySelector?: number;
  epcLength?: number;
  tidLength?: number;
  userLength?: number;
  selectionMaskCount?: number;
  selectionMaskMaxLength?: number;
  customOperationMaxLength?: number;
  includeAccessPwd?: boolean;
  includeCRC?: boolean;
  includeKillPwd?: boolean;
  includePC?: boolean;
  includeXPC?: boolean;
}

export class RuntimeConfiguration implements RuntimeConfigurationIntf {
  private static readonly KILL_PWD = 0b1;
  private static readonly ACCESS_PWD = 0b10;
  private static readonly CRC = 0b100;
  private static readonly PC = 0b1000;
  private static readonly XPC = 0b10000;

  tagsInField?: number;
  memorySelector?: number;
  epcLength?: number;
  tidLength?: number;
  userLength?: number;
  selectionMaskCount?: number;
  selectionMaskMaxLength?: number;
  customOperationMaxLength?: number;

  constructor(
    source: RuntimeConfigurationIntf,
  ) {
    Object.assign(this, source);
  }
  private setMemValue(include: boolean, value: number) {
    if (include) {
      this.memorySelector |= value;
    } else {
      this.memorySelector &= ~value;
    }
  }

  private getMemValue(value: number) {
    return (this.memorySelector & value) > 0;
  }

  get includeXPC() {
    return this.getMemValue(RuntimeConfiguration.XPC);
  }

  set includeXPC(includeXPC: boolean) {
    this.setMemValue(includeXPC, RuntimeConfiguration.XPC);
  }

  get includePC() {
    return this.getMemValue(RuntimeConfiguration.PC);
  }

  set includePC(includePC: boolean) {
    this.setMemValue(includePC, RuntimeConfiguration.PC);
  }

  get includeCRC() {
    return this.getMemValue(RuntimeConfiguration.CRC);
  }

  set includeCRC(includeCRC: boolean) {
    this.setMemValue(includeCRC, RuntimeConfiguration.CRC);
  }

  get includeAccessPwd() {
    return this.getMemValue(RuntimeConfiguration.ACCESS_PWD);
  }

  set includeAccessPwd(includeAccessPwd: boolean) {
    this.setMemValue(includeAccessPwd, RuntimeConfiguration.ACCESS_PWD);
  }

  get includeKillPwd() {
    return this.getMemValue(RuntimeConfiguration.KILL_PWD);
  }

  set includeKillPwd(includeKillPwd: boolean) {
    this.setMemValue(includeKillPwd, RuntimeConfiguration.KILL_PWD);
  }
}

File: src/main/java/havis/app/modbus/reader/rest/data/RuntimeConfiguration.java
package havis.app.modbus.reader.rest.data;

public class RuntimeConfiguration {

	private static int KILL_PWD = 0b1;
	private static int ACCESS_PWD = 0b10;
	private static int CRC = 0b100;
	private static int PC = 0b1000;
	private static int XPC = 0b10000;

	private int tagsInField;
	private short memorySelector;
	private int epcLength;
	private int tidLength;
	private int userLength;
	private int selectionMaskCount;
	private int selectionMaskMaxLength;
	private int customOperationMaxLength;

	public int getTagsInField() {
		return tagsInField;
	}

	public void setTagsInField(int tagsInField) {
		this.tagsInField = tagsInField;
	}

	public short getMemorySelector() {
		return memorySelector;
	}

	public void setMemorySelector(short memorySelector) {
		this.memorySelector = memorySelector;
	}

	private void setMemValue(boolean include, int value) {
		if (include) {
			this.memorySelector |= value;
		} else {
			this.memorySelector &= ~value;
		}
	}

	private boolean getMemValue(int value) {
		return (this.memorySelector & value) > 0;
	}

	public boolean isIncludeXPC() {
		return getMemValue(XPC);
	}

	public void setIncludeXPC(boolean includeXPC) {
		setMemValue(includeXPC, XPC);
	}

	public boolean isIncludePC() {
		return getMemValue(PC);
	}

	public void setIncludePC(boolean includePC) {
		setMemValue(includePC, PC);
	}

	public boolean isIncludeCRC() {
		return getMemValue(CRC);
	}

	public void setIncludeCRC(boolean includeCRC) {
		setMemValue(includeCRC, CRC);
	}

	public boolean isIncludeAccessPwd() {
		return getMemValue(ACCESS_PWD);
	}

	public void setIncludeAccessPwd(boolean includeAccessPwd) {
		setMemValue(includeAccessPwd, ACCESS_PWD);
	}

	public boolean isIncludeKillPwd() {
		return getMemValue(KILL_PWD);
	}

	public void setIncludeKillPwd(boolean includeKillPwd) {
		setMemValue(includeKillPwd, KILL_PWD);
	}

	public int getEpcLength() {
		return epcLength;
	}

	public void setEpcLength(int epcLength) {
		this.epcLength = epcLength;
	}

	public int getTidLength() {
		return tidLength;
	}

	public void setTidLength(int tidLength) {
		this.tidLength = tidLength;
	}

	public int getUserLength() {
		return userLength;
	}

	public void setUserLength(int userLength) {
		this.userLength = userLength;
	}

	public int getSelectionMaskCount() {
		return selectionMaskCount;
	}

	public void setSelectionMaskCount(int selectionMaskCount) {
		this.selectionMaskCount = selectionMaskCount;
	}

	public int getSelectionMaskMaxLength() {
		return selectionMaskMaxLength;
	}

	public void setSelectionMaskMaxLength(int selectionMaskMaxLength) {
		this.selectionMaskMaxLength = selectionMaskMaxLength;
	}

	public int getCustomOperationMaxLength() {
		return customOperationMaxLength;
	}

	public void setCustomOperationMaxLength(int customOperationMaxLength) {
		this.customOperationMaxLength = customOperationMaxLength;
	}

}
